### PR TITLE
Issue#205 Аdd filter on deactivateUser method

### PR DESCRIPTION
**Pull request by veniamin-aveltsev**
…vating user has role Admin.

dev
## ZenHub

* [Main ZenHub ticket](https://app.zenhub.com/workspaces/home-project-5f7b77ff8db73a001cb17009)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: home-service/src/main/java/com/softserveinc/ita/homeproject/homeservice/service/impl/UserServiceImpl.java [hunk 1]
```diff
@@ -116,6 +118,9 @@ public UserDto getUserById(Long id) {
     public void deactivateUser(Long id) {
         User toDelete = userRepository.findById(id).filter(User::getEnabled)
             .orElseThrow(() -> new NotFoundHomeException(String.format(USER_NOT_FOUND_FORMAT, id)));
+        if (toDelete.getRoles().contains(roleRepository.findByName(ADMIN_ROLE))) {
```
Comment (kh0ma) [inline home-service/src/main/java/com/softserveinc/ita/homeproject/homeservice/service/impl/UserServiceImpl.java]: Can we cover it via API test?

**Review by kh0ma — APPROVED** [review]
LGTM with the suggested change.

**Review by likeRewca — APPROVED** [review]

File: home-service/src/main/java/com/softserveinc/ita/homeproject/homeservice/service/impl/UserServiceImpl.java (hunk 1 above)
Comment (veniamin-aveltsev) [reply home-service/src/main/java/com/softserveinc/ita/homeproject/homeservice/service/impl/UserServiceImpl.java]: Okay, I will add a test for  сhecking the removal of the admin

Comment (veniamin-aveltsev) [reply home-service/src/main/java/com/softserveinc/ita/homeproject/homeservice/service/impl/UserServiceImpl.java]: In order to remove a test user, we need to assign him some role,
but we have no way in which the role can be changed
Therefore, the test with the removal of the user with the role of the administrator cannot be done yet

**Review by tex1988 — APPROVED** [review]

**Review by IRoss-git — APPROVED** [review]

**Pull request merged**